### PR TITLE
CHK-573 Add optional dueDate in PaymentRequestsGetResponse

### PR DESCRIPTION
**Pull request by infantesimone**
#### List of Changes

- add optional `dueDate` in `PaymentRequestsGetResponse`

#### Motivation and Context

The goal of this PR is to add `dueDate` in verify step, according to https://raw.githubusercontent.com/pagopa/pagopa-api/develop/nodo/nodeForPsp.wsdl.

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api-spec/api-for-checkout.yaml [hunk 1]
```diff
@@ -143,7 +143,31 @@ definitions:
...
+        type: string
+        minLength: 1
+        maxLength: 140
+      enteBeneficiario:
+        $ref: '#/definitions/EnteBeneficiario'
+      spezzoniCausaleVersamento:
+        $ref: '#/definitions/SpezzoniCausaleVersamento'
+      dueDate:
+        type: string
+        format: date
```
Comment (giovanniberti) [inline api-spec/api-for-checkout.yaml:166]: Should be `date-time`

File: api-spec/api-for-io.yaml [hunk 2]
```diff
@@ -297,6 +297,10 @@ definitions:
         $ref: '#/definitions/EnteBeneficiario'
       spezzoniCausaleVersamento:
         $ref: '#/definitions/SpezzoniCausaleVersamento'
+      dueDate:
+        type: string
+        format: date
```
Comment (giovanniberti) [inline api-spec/api-for-io.yaml:302]: Same as above

File: api-spec/api-pagopa-proxy.yaml [hunk 3]
```diff
@@ -507,6 +507,10 @@ definitions:
         $ref: '#/definitions/EnteBeneficiario'
       spezzoniCausaleVersamento:
         $ref: '#/definitions/SpezzoniCausaleVersamento'
+      dueDate:
+        type: string
+        format: date
```
Comment (giovanniberti) [inline api-spec/api-pagopa-proxy.yaml:512]: Ditto

File: api-spec/api-for-checkout.yaml (hunk 1 above)
Comment (infantesimone) [reply api-spec/api-for-checkout.yaml]: 🙌

File: api-spec/api-for-io.yaml (hunk 2 above)
Comment (infantesimone) [reply api-spec/api-for-io.yaml]: 🙌

File: api-spec/api-pagopa-proxy.yaml (hunk 3 above)
Comment (infantesimone) [reply api-spec/api-pagopa-proxy.yaml]: 🙌

**Review by MarcoAGreco — APPROVED** [review]
👌

File: src/utils/__tests__/PaymentConverter.test.ts [hunk 4]
```diff
@@ -918,4 +919,45 @@ describe("getPaymentNoticeNumberAsString", () => {
...
+      MockedData.verifyPaymentNoticeWithDueDate, 
+      "8447a9f0746811e89a8d5d4209060ab0" as CodiceContestoPagamento
+    );
+    expect(isRight(errorOrVerifyPaymentNoticeResponse)).toBeTruthy();
+    if (!isRight(errorOrVerifyPaymentNoticeResponse)) {
+      return;
+    }
+
+    const dueDate : Date | undefined = errorOrVerifyPaymentNoticeResponse.right.dueDate
+    ? errorOrVerifyPaymentNoticeResponse.right.dueDate : undefined;
```
Comment (giovanniberti) [inline src/utils/__tests__/PaymentConverter.test.ts:938]: Duplicated `errorOrVerifyPaymentNoticeResponse.right.dueDate`?

File: src/utils/__tests__/PaymentConverter.test.ts [hunk 5]
```diff
@@ -918,4 +919,45 @@ describe("getPaymentNoticeNumberAsString", () => {
...
+      MockedData.verifyPaymentNoticeWithoutDueDate, 
+      "8447a9f0746811e89a8d5d4209060ab0" as CodiceContestoPagamento
+    );
+    expect(isRight(errorOrVerifyPaymentNoticeResponse)).toBeTruthy();
+    if (!isRight(errorOrVerifyPaymentNoticeResponse)) {
+      return;
+    }
+
+    const dueDate : Date | undefined = errorOrVerifyPaymentNoticeResponse.right.dueDate
+    ? errorOrVerifyPaymentNoticeResponse.right.dueDate : undefined;
```
Comment (giovanniberti) [inline src/utils/__tests__/PaymentConverter.test.ts:957]: Duplicated `errorOrVerifyPaymentNoticeResponse.right.dueDate`?

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**